packages/example/src/components/EditWallet/index.tsx

```diff
@@ -0,0 +1,267 @@
+import { useEffect, useMemo, useRef, useState } from 'react';
+import { observer } from 'mobx-react-lite';
+import { useAppStore } from '../../mobx';
+import { Button, H3, Message, MessageType, Modal } from '../../kits';
+import RecoveryKey from '../Lightning/CreateWallet/RecoveryKey';
+import EditIcon from '../Icons/EditIcon';
+import { WalletType } from '../../interface';
+import { EditWalletContainer } from './styles';
+import { List } from '../../kits/List';
+import Divider from '../../kits/Divider';
+import LightningWallet from '../../mobx/lightningWallet';
+import { Instance } from 'mobx-state-tree';
+import { getLNWalletData, GetLNWalletDataKey } from '../../lib/snap';
+
+interface EditWalletProps {
+  open: boolean;
+  id: string | null;
+  walletType: WalletType;
+  close: () => void;
+  showSelfModal: (args: { id: string; type: WalletType }) => void;
+}
+
+type LightningWallet = Instance<typeof LightningWallet>;
+
+type WalletInfo = { name: string } & Partial<LightningWallet>;
+
+const walletTypeTextMap = {
+  [WalletType.BitcoinWallet]: 'Bitcoin On Chain Wallet',
+  [WalletType.LightningWallet]: 'Lightning',
+};
+
+const EditWallet = observer(
+  ({ open, id, walletType, close, showSelfModal }: EditWalletProps) => {
+    const { lightning } = useAppStore();
+    const parentNode = useRef<HTMLDivElement>(null);
+    const [showConfirm, setShowConfirm] = useState(false);
+    const [errorMessage, setErrorMessage] = useState('');
+    const modalRef = useRef<any>(null);
+    const [pendingSnap, setPendingSnap] = useState(false);
+
+    const {
+      current,
+      lightning: { getWalletByUserId, updateWallet: updateLightningWallet },
+      user: { name: bitcoinWalletName, updateUser },
+      switchToWallet,
+    } = useAppStore();
+
+    const [recoveryKey, setRecoveryKey] = useState({
+      open: false,
+      value: '',
+    });
+
+    const [backWalletInfo, setBackWalletInfo] = useState<WalletInfo>();
+    const [walletInfo, setWalletInfo] = useState<WalletInfo>();
+
+    useEffect(() => {
+      if (!id) return;
+      if (walletType === WalletType.LightningWallet) {
+        setWalletInfo(getWalletByUserId(id));
+        setBackWalletInfo(getWalletByUserId(id));
+      } else {
+        setWalletInfo({
+          name: bitcoinWalletName,
+        });
+        setBackWalletInfo({
+          name: bitcoinWalletName,
+        });
+      }
+    }, [id]);
+
+    const shouldEnableSaveButton = useMemo(() => {
+      return JSON.stringify(walletInfo) === JSON.stringify(backWalletInfo);
+    }, [walletInfo, backWalletInfo]);
+
+    function onDeleteWallet() {
+      if (!id) {
+        console.error('id not found');
+        return;
+      }
+      if (walletType === WalletType.LightningWallet) {
+        lightning.removeWallet(id);
+        switchToWallet(WalletType.BitcoinWallet, current?.xpub);
+      }
+      onCloseConfirm();
+      modalRef.current?.onClose();
+    }
+
+    function onShowDeleteConfirm() {
```

Comment:
prefer `showDeleteConfirmModal`, `on` is usually used as callback, e.g. `onClick`, `onTrigger`.